Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,465 @@
+/-
+Copyright (c) 2024 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Josha Dekker
+-/
+import Mathlib.MeasureTheory.Measure.Regular
+import Mathlib.Topology.Metrizable.Uniformity
+import Mathlib.Analysis.SpecificLimits.Basic
+
+/-!
+# (Pre-)tight measures
+The key definition of interest here is that of tight measures, `IsTight`. We first introduce two
+weaker notions, `IsSeparable` and `IsPretight`, which are equivalent on complete metric spaces. We
+provide some basic API for these notions and prove Ulam's tightness theorem
+(`of_isSeparableSpace_complete_uniform`) and its strengthening `of_isSeparable_complete_uniform`.
+
+## Main definitions
+* `IsSeparable`: A measure `μ` is separable if there is a separable set `S` such that
+  `μ S = μ Set.univ`.
+* `IsPretight`: A measure `μ` is pre-tight if for all `0 < ε`, there exists `K` totally bounded such
+  that `μ Kᶜ ≤ ε`.
+* `IsTight`: A measure `μ` is tight if for all `0 < ε`, there exists `K` compact such that
+  `μ Kᶜ ≤ ε`. This is defined in terms of filters. `IsTight.iff_compact_sets` establishes
+  equivalence with the usual definition.
+
+## Main statements
+
+* `of_isSeparableSpace_complete_uniform`: Ulam's tightness theorem: a finite measure on a complete
+  separable metric space is tight.
+* `of_isSeparable_complete_uniform`: A strengthening of Ulam's tightness theorem: a finite,
+  separable measure on a complete metric space is tight.
+* `of_innerRegular`: Every finite, inner-regular measure on a T2 space is tight.
+
+## Notation
+
+
+## Implementation details
+
+
+-/
+open Topology Filter Uniformity Uniform MeasureTheory Set
+open scoped ENNReal
+
+namespace MeasureTheory
+
+variable {α ι : Type*}
+
+variable [MeasurableSpace α] {μ : Measure α}
+
+/-- For a finite measure `μ`, we can extract from a countable cover that has full measure, a finite
+cover of accumulated sets that has `ε`-almost full measure. -/
+private lemma almost_cover_has_approx_accumulate [MeasurableSpace α] {μ : Measure α}
+    [IsFiniteMeasure μ] (K : ℕ → Set α) (h : μ (⋃ n, K n) = μ Set.univ) :
+    ∀ ε > 0, ∃ n, μ (Set.Accumulate K n) ≥ μ Set.univ - ε := by
```

Comment:
```suggestion
    ∀ ε > 0, ∃ n, μ Set.univ - ε ≤ μ (Set.Accumulate K n) := by
```
Please use only inequalities signs "less than" or "less than or equal to", for coherence everywhere in the library (except in quantifiers like your `∀ ε > 0`, where it's not possible to swap directions).